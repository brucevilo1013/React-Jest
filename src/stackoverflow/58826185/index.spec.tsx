import React from 'react';
import { Component } from './';
import { mount } from 'enzyme';

describe('Component', () => {
  test('should handle change event', async () => {
    const logSpy = jest.spyOn(console, 'log');
    const wrapper = mount(<Component></Component>);
    expect(logSpy).toBeCalledWith('');
    wrapper.find('input').simulate('change', { target: { value: 'UT' } });
    expect(logSpy).toBeCalledWith('UT');
    expect(wrapper).toMatchSnapshot();
    logSpy.mockRestore();
  });
});
