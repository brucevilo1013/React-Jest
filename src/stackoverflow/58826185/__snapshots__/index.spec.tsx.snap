// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component should handle change event 1`] = `
<Component>
  <View>
    <div>
      <CustomComponent
        onChange={[Function]}
      >
        <input
          onChange={[Function]}
        />
      </CustomComponent>
    </div>
  </View>
</Component>
`;
